job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -531,6 +534,36 @@ class WebApi(system: ActorSystem,
   override def timeoutRoute: Route =
     complete(500, errMap("Request timed out. Try using the /jobs/<jobID>, /jobs APIs to get status/results"))
 
+  def determineProxyUser(aConfig: Config,
+                         authInfo: AuthInfo,
+                         contextName: String): (String, Config) = {
+    if (config.getBoolean("shiro.authentication") && config.getBoolean("shiro.use-as-proxy-user")) {
+      //proxy-user-param is ignored and the authenticated user name is used
+      val config = aConfig.withValue(SparkJobUtils.SPARK_PROXY_USER_PARAM,
+        ConfigValueFactory.fromAnyRef(authInfo.toString))
+      (authInfo.toString + "_" + contextName, config)
+    } else {
+      (contextName, aConfig)
+    }
+  }
+
+  def removeProxyUserPrefix(authInfo: AuthInfo, contextNames: Seq[String]): Seq[String] = {
+    if (config.getBoolean("shiro.authentication") && config.getBoolean("shiro.use-as-proxy-user")) {
+      val userNamePrefix = authInfo.toString + "_"
+      contextNames map { cName =>
+        {
+          if (cName.startsWith(userNamePrefix)) {
```

Comment:
mmmm _ is not a valid character for user names right?... I can't have fernando_otero as my username?

I think this restriction has to be documented or we might need to use a less likely separator than '_'